src/handler.rs

```diff
@@ -9,11 +9,22 @@ pub enum Outcome {
     Response(Vec<String>),
 }
 
+#[derive(Debug)]
+pub enum Error {
+    MissingSource,
+    BadArgument(u8),
```

Comment:
This variant could use a string to describe why the argument is bad. For instance, this is currently returned in a place where previously the string "invalid sid" was returned.